### PR TITLE
4927-One Inch config added to chain-abstraction SDK

### LINKED ISSUE #4927: Chain Abstraction SDK 1inch custom API key
**Describe the bug**

Follow-up task from https://github.com/connext/monorepo/issues/4879

Allow users to provide a custom API key for 1Inch.

**Additional context**

### PR DESCRIPTION
**Pull request by prathmeshkhandelwal1**
## Description

Added a Authorization config for One Inch API

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Related Issue(s)

Fixes #4927

### CODE REVIEW

**Review by rhlsthrm — DISMISSED** [review]

File: packages/agents/chain-abstraction/src/helpers/swapdata.ts [hunk 1]
```diff
@@ -11,6 +11,15 @@ export type OriginSwapDataCallbackArgs = {
...
   fromAddress: string;
+  config?:
+    | {
+        customURL: string;
+        apiKey?: string;
+      }
+    | {
+        customURL?: undefined;
+        apiKey: string;
+      };
```
Comment (wanglonghong) [inline packages/agents/chain-abstraction/src/helpers/swapdata.ts:22]: I guess we can make it better like 

```suggestion
  config?:
   {
        customURL?: string;
        apiKey?: string;
   }
```

**Review by wanglonghong — CHANGES_REQUESTED** [review]

File: packages/agents/chain-abstraction/src/libs/origin/swapandxcall.ts [hunk 2]
```diff
@@ -20,10 +20,20 @@ import { OriginSwapDataFns, OriginSwapperPerDomain, DEPLOYED_ADDRESSES } from ".
...
   signerAddress: string,
+  config?:
+    | {
+        customURL: string;
+        apiKey?: string;
+      }
+    | {
+        customURL?: undefined;
+        apiKey: string;
+      },
```
Comment (wanglonghong) [inline packages/agents/chain-abstraction/src/libs/origin/swapandxcall.ts:36]: same comments as above

File: packages/agents/chain-abstraction/src/libs/origin/swapandxcall.ts [hunk 3]
```diff
@@ -158,6 +170,15 @@ export const calculateRouteForSwapAndXCall = async (
...
   fromAddress: string,
+  config?:
+    | {
+        customURL: string;
+        apiKey?: string;
+      }
+    | {
+        customURL?: undefined;
+        apiKey: string;
+      },
```
Comment (wanglonghong) [inline packages/agents/chain-abstraction/src/libs/origin/swapandxcall.ts:181]: same comments as above

File: packages/agents/chain-abstraction/src/helpers/swapdata.ts [hunk 4]
```diff
@@ -38,14 +47,33 @@ export const getOriginSwapDataForUniV3 = async (_args: OriginSwapDataCallbackArg
  * including a function signature for the 1inch aggregator.
  */
 export const getOriginSwapDataForOneInch = async (args: OriginSwapDataCallbackArgs): Promise<string> => {
+  if (!args.config) throw new Error("No Authorizartion config provided for One Inch.");
```
Comment (wanglonghong) [inline packages/agents/chain-abstraction/src/helpers/swapdata.ts:50]: The following lines says both params: `customURL` and `apiKey` can be optional. then we can update it like this
```
  config?: {
    customURL?: string;
    apiKey?: string;
  }; 
```
If both are not provided here, throw like above: `if (!args.config) throw new ...`
```
if (!args.config.customURL && !args.config.apiKey) throw new Error("...")
```

File: packages/agents/chain-abstraction/src/helpers/swapdata.ts [hunk 5]
```diff
@@ -38,14 +47,34 @@ export const getOriginSwapDataForUniV3 = async (_args: OriginSwapDataCallbackArg
...
```
Comment (rhlsthrm) [inline packages/agents/chain-abstraction/src/helpers/swapdata.ts:50]:
```suggestion
  if (!args.config) throw new Error("No Authorization config provided for 1Inch.");
```

**Review by rhlsthrm — CHANGES_REQUESTED** [review]

**Review by rhlsthrm — APPROVED** [review]

**Review by wanglonghong — APPROVED** [review]

**Pull request merged**